c/parallel/src/util/indirect_arg.h

```diff
@@ -25,7 +25,7 @@ struct indirect_arg_t
   {}
 
   indirect_arg_t(cccl_op_t& op)
-      : ptr(op.type == cccl_op_kind_t::CCCL_STATEFUL ? op.state : this)
+      : ptr(op.type == cccl_op_kind_t::CCCL_STATEFUL ? op.state : &op)
```

Comment:
Just curious: was this change necessary because you ran into a problem with current implementation? The trouble with using `&op` is that that address may become invalid one `cccl_op_t` struct passed by reference is destroyed. Using `this` the address stored in `ptr` is always valid, but not useful.

Reply:
@NaderAlAwar If `LargeSegmentsSelector` is annotated with `[[nodiscard]]`  `this` would not be temporary. The only scenario I can think of where `this` becomes a dangling pointer is when the struct `indirect_arg_t` returned by `LargeSegmentsSelector` is never assigned to an l-value.

Reply:
Let's punt this to a another PR